lib/macho/load_commands.rb

```diff
@@ -141,6 +143,8 @@ module LoadCommands
       :LC_LINKER_OPTIMIZATION_HINT => "LinkeditDataCommand",
       :LC_VERSION_MIN_TVOS => "VersionMinCommand",
       :LC_VERSION_MIN_WATCHOS => "VersionMinCommand",
+      :LC_NOTE => "LoadCommand",
+      :LC_BUILD_VERSION => "VersionMinCommand",
```

Comment:
This probably needs a new class, based on the corresponding structure in LLVM's Mach-O parser:

```c
    struct build_version_command {
        uint32_t    cmd;        /* LC_BUILD_VERSION */
        uint32_t    cmdsize;    /* sizeof(struct build_version_command) plus */
        /* ntools * sizeof(struct build_tool_version) */
        uint32_t    platform;   /* platform */
        uint32_t    minos;      /* X.Y.Z is encoded in nibbles xxxx.yy.zz */
        uint32_t    sdk;        /* X.Y.Z is encoded in nibbles xxxx.yy.zz */
        uint32_t    ntools;     /* number of tool entries following this */
    };
```

Reply:
(For comparison, `version_min_command` has just two fields: `version` and `sdk`).

Reply:
Change made and tested with a binary. No `build_tool_version` yet.

Reply:
`TwoLevelHintsCommand` might be a good reference for adding `build_tool_version`, since it encapsulates its own table/array structure.

Reply:
OK, updated. We weren't handed an offset to read the table from so I'm making the assumption here that it's the offset following `view.offset + 24`.

While we're not given much info about the tool structure, I am seeing what looks like correct information from what we do have:

```ruby
irb(main):005:0> command.platform #=> 1, e.g. macOS
irb(main):006:0> command.minos_string #=> "10.14.0"
irb(main):007:0> command.sdk_string #=> "10.14.0"
irb(main):008:0> command.tool_entries.tools.first.tool #=> 3, e.g. ld
irb(main):009:0> command.tool_entries.tools.first.version #=> 26738945, format of this string not documented
```

Reply:
> We weren't handed an offset to read the table from so I'm making the assumption here that it's the offset following view.offset + 24.

Yeah, I think this is a reasonable assumption.